test/quoteSpec.js

```diff
@@ -0,0 +1,247 @@
+'use strict'
+
+const chai = require('chai')
+const moment = require('moment')
+const assert = chai.assert
+const ILQP = require('..').ILQP
+const Packet = require('../src/utils/packet')
+const MockPlugin = require('./mocks/mockPlugin')
+const expect = chai.expect
+const chaiAsPromised = require('chai-as-promised')
+chai.use(chaiAsPromised)
+
+describe('ILQP', function () {
+  beforeEach(function () {
+    this.plugin = new MockPlugin()
+    // quote response
+    this.id = '85e04e5c-2357-4033-ac5b-251ce97faf33'
+    this.response = {
+      data: {
+        id: this.id,
+        method: 'quote_response',
+        data: {
+          source_amount: '1',
+          destination_amount: '1',
+          source_connector_account: 'test.example.connie',
+          source_expiry_duration: '0'
+        }
+      }
+    }
+  })
+
+  describe('quote', function () {
+    beforeEach(function () {
+      this.params = {
+        sourceAddress: 'test.example.alice',
+        destinationAddress: 'test.local.bob',
+        sourceAmount: '1',
+        sourceExpiryDuration: '0',
+        destinationExpiryDuration: '0',
+        connectors: [ 'test.example.connie' ]
+      }
+      this.result = {
+        sourceAmount: '1',
+        destinationAmount: '1',
+        connectorAccount: 'test.example.connie',
+        sourceExpiryDuration: '0'
+      }
+
+      this.plugin.sendMessage = (msg) => {
```

Comment:
The problem I see with using this custom setup instead of something like sinon is that you're not actually checking what the arguments are that are passed to this mock